src/evtx_parser.rs

```diff
@@ -70,132 +153,141 @@ impl<T: ReadSeek> EvtxParser<T> {
         EvtxChunkData::new(chunk_data)
     }
 
-    #[cfg(feature = "multithreading")]
-    pub fn parallel_records(self) -> IterRecords<T> {
-        IterRecords {
-            header: self.header,
-            data: self.data,
+
+    /// Return an iterator over all the chunks.
+    /// Each chunk supports iterating over it's records in their un-serialized state
+    /// (before they are converted to XML or JSON).
+    pub fn chunks(&mut self) -> IterChunks<T> {
+        IterChunks {
+            parser: self,
             current_chunk_number: 0,
-            chunk_records: None,
-            num_threads: current_num_threads(),
-            exhausted: false,
         }
     }
 
-    pub fn records(self) -> IterRecords<T> {
-        IterRecords {
-            header: self.header,
-            data: self.data,
-            current_chunk_number: 0,
-            chunk_records: None,
-            num_threads: 1,
-            exhausted: false,
+    /// Return an iterator over all the records.
+    /// Records will be serialized using the given `BinXmlOutput`.
+    pub fn serialized_records<'a, O: 'a + BinXmlOutput<Vec<u8>>>(&'a mut self) -> impl Iterator<Item=Result<SerializedEvtxRecord, Error>> + 'a {
+        IterSerializedRecords::<'a, T, O> {
+            chunks: self.chunks(),
+            current_chunk_records: None,
+            _phantom: PhantomData,
         }
     }
-}
 
-impl<T: ReadSeek> IterRecords<T> {
-    fn allocate_chunks(&mut self) -> Result<(), Error> {
-        let mut chunks = vec![];
-
-        // Some dirty samples contains additional chunks (further than the marked number).
-        // So instead of trusting the file header, we opportunistically try to read a bit further.
-        // If we fail, we terminate iteration.
-        for _ in 0..self.num_threads {
-            info!("Allocating new chunk {}", self.current_chunk_number);
-            match EvtxParser::allocate_chunk(&mut self.data, self.current_chunk_number) {
-                Ok(chunk) => {
-                    chunks.push(chunk);
-                    self.current_chunk_number += 1;
-                }
-                Err(_) => {
-                    info!("Next block does not contain a valid chunk. terminating iteration.");
-                    self.exhausted = true;
-                    break;
+    /// Return an iterator over all the records.
+    /// Records will be serialized using the given `BinXmlOutput`.
+    #[cfg(feature = "multithreading")]
+    pub fn serialized_records_parallel<O: BinXmlOutput<Vec<u8>>>(&mut self) -> impl Iterator<Item=Result<SerializedEvtxRecord, Error>>
```

Comment:
it seems that this isn't called anywhere, and it should also consider "num_threads".

Wasn't this supposed to be the same as `pub fn serialized_records` when cfg multithreading is enabled?